tests/unit/amo/components/TestAddonBadges.js

```diff
@@ -42,18 +50,113 @@ describe(__filename, () => {
     expect(root.find(Badge)).toHaveLength(0);
   });
 
-  it('displays a badge when the addon is featured', () => {
+  it('displays a badge when the addon is recommended', () => {
+    const { store } = dispatchClientMetadata({
+      clientApp: CLIENT_APP_FIREFOX,
+    });
+
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      is_recommended: true,
+      type: ADDON_TYPE_EXTENSION,
+    });
+    const root = shallowRender({
+      _config: getFakeConfig({
+        enableFeatureRecommendedBadges: true,
+      }),
+      addon,
+      store,
+    });
+
+    expect(root.find(RecommendedBadge)).toHaveLength(1);
+  });
+
+  it('does not display a recommended badge on Android', () => {
+    const { store } = dispatchClientMetadata({
+      clientApp: CLIENT_APP_ANDROID,
+    });
+
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      is_recommended: true,
+      type: ADDON_TYPE_EXTENSION,
+    });
+    const root = shallowRender({
+      _config: getFakeConfig({
+        enableFeatureRecommendedBadges: true,
+      }),
+      addon,
+      store,
+    });
+
+    expect(root.find(RecommendedBadge)).toHaveLength(0);
+  });
+
+  it('does not display a recommended badge when the feature is disabled', () => {
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      is_recommended: true,
+      type: ADDON_TYPE_EXTENSION,
+    });
+    const root = shallowRender({
+      _config: getFakeConfig({
+        enableFeatureRecommendedBadges: false,
+      }),
+      addon,
+    });
+
+    expect(root.find(RecommendedBadge)).toHaveLength(0);
+  });
+
+  it('displays a badge for an extension when the addon is featured and the recommended feature is off', () => {
     const addon = createInternalAddon({
       ...fakeAddon,
       is_featured: true,
       type: ADDON_TYPE_EXTENSION,
     });
-    const root = shallowRender({ addon });
+    const root = shallowRender({
+      _config: getFakeConfig({
+        enableFeatureRecommendedBadges: false,
+      }),
+      addon,
+    });
 
     expect(root.find(Badge)).toHaveProp('type', 'featured');
     expect(root.find(Badge)).toHaveProp('label', 'Featured Extension');
   });
 
+  it('displays a badge for a theme when the addon is featured and the recommended feature is on', () => {
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      is_featured: true,
+      type: ADDON_TYPE_THEME,
+    });
+    const root = shallowRender({
+      _config: getFakeConfig({
+        enableFeatureRecommendedBadges: true,
+      }),
+      addon,
+    });
+
+    expect(root.find(Badge)).toHaveProp('type', 'featured');
+  });
+
+  it('does not display a badge for an extension when the addon is featured and the recommended feature is on', () => {
```

Comment:
What if the extension is both featured and recommended? Do we need to write a test case for that?

Reply:
I don't think we do. The two are unrelated, i.e., the `is_recommended` flag has nothing to do with whether we show the Featured badge, and vice versa. It seems like an unnecessary test to me.